pilot/pkg/networking/util/util.go

```diff
@@ -480,18 +480,22 @@ func MergeAnyWithAny(dst *any.Any, src *any.Any) (*any.Any, error) {
 }
 
 // BuildLbEndpointMetadata adds metadata values to a lb endpoint
-func BuildLbEndpointMetadata(network network.ID, tlsMode, workloadname, namespace string,
+func BuildLbEndpointMetadata(networkID network.ID, tlsMode, workloadname, namespace string,
 	clusterID cluster.ID, labels labels.Instance) *core.Metadata {
-	if network == "" && (tlsMode == "" || tlsMode == model.DisabledTLSModeLabel) && !features.EndpointTelemetryLabel {
+	if networkID == "" && (tlsMode == "" || tlsMode == model.DisabledTLSModeLabel) && !features.EndpointTelemetryLabel {
 		return nil
 	}
 
 	metadata := &core.Metadata{
 		FilterMetadata: map[string]*pstruct.Struct{},
 	}
 
-	if network != "" {
-		addIstioEndpointLabel(metadata, "network", &pstruct.Value{Kind: &pstruct.Value_StringValue{StringValue: network.String()}})
+	if networkID != "" {
+		addIstioEndpointLabel(metadata, "network", &pstruct.Value{Kind: &pstruct.Value_StringValue{StringValue: networkID.String()}})
+	}
+	// TODO(nmittler): Is there any concern with adding an extra label?
```

Comment:
Yes. I don't have the numbers off hand but I think even adding one label leads to a big impact on EDS cost (since there are so many endpoints). Why do we need it ?

Actually confused why we have `network` as well.. We should check the benchmark results

Reply:
Only one way to find out: https://github.com/istio/istio/pull/33717

Reply:
@stevenctl ^^

I suspect the reason for `network` was similar to why I added `cluster` ... some information from `IstioEndpoint` is lost post transformation to the Envoy structs.

Reply:
We only need it for this in-control-plane logic. If we can fetch it after the fact (find the original istio endpoints) or if we can strip it after this is used that would be good. 

For the "stripping after use" case, we probably want to only add it conditionally, based on MultiNetworkEnabled

Reply:
I've removed `network` and `cluster` from LBEndpoint metadata.